keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,832 @@
-	
-Upon success, the `BucketAccess` instance is ready and the app workload can access backend storage.
+###### 1. Admin creates a Bucket API object
 
-### Static Provisioning
-"Driverless" allows the existing workloads to make use of COSI without the need for vendors to create drivers. The following steps show the details of the workflow:
-+ Admin creates `Bucket` instance that references an existing backend bucket.
-+ User creates a `BucketRequest` (BR) that references this `Bucket`.
-+ User creates `BucketAccessRequest` (BAR) that references their BR, a `BucketAccessClass`, and their secret containing access tokens.
-+ COSI detects the existence of the BAR and follows the [grant access](#grant-bucket-access) steps.
-+ COSI detects the existence of the BAR, BA and executes a validation process.
-+ As a validation step, COSI ensures that the `BucketRequest.bucketInstanceName` matches the `BucketAccessRequest.bucketAccessName.bucketInstanceName`. In other words, the two related `Bucket` references match, meaning that the BR and BAR->BA point to the same `Bucket`.
+When a Bucket is manually created, and has its `bucketID` set, then COSI assumes that this Bucket has already been created. 
 
----
+```
+    Bucket - br-$uuid
+    |-------------------------------------------------|
+    | name: bucketName123                             |
+    | spec:                                           |
+    |   bucketID: bucketname123                       |
+    |   protocol: s3                                  |
+    |   parameters:                                   |
+    |     key: value                                  |
+    |   provisioner: s3.amazonaws.com                 |
+    |-------------------------------------------------|
+```
 
-# gRPC Definitions
+###### 2. User creates BucketAccess to generate credentials for that bucket
 
-There is one service defined by COSI - `Provisioner`. This will need to be satisfied by the vendor-provisioner in order to be COSI-compatible
+Unlike the BucketAccess for COSI created bucket, this BucketClaim should directly reference the Bucket instead of a BucketClaim
 
 ```
-service Provisioner {
-    rpc ProvisionerGetInfo (ProvisionerGetInfoRequest) returns (ProvisionerGetInfoResponse) {}
+    BucketAccess - bac2
+    |-------------------------------|
+    | spec:                         |
+    |   bucketName: bucketName123   |
+    |                               |
+    |-------------------------------|
+```
+This BucketClaim can be specified in pod definition as described in the above sections.
+
+Note that, as of the alpha version of COSI, there is no authorization mechanism to restrict users who can refer to buckets created by other namesapces. In the future, we are working on a design based on `referencePolicy`, a new development in K8s, meant to be a general approach to sharing resources across namespaces.
```

Comment:
On another note, I have some strong objections to allowing anyone to specify any bucket without a way for an admin to restrict access. I think we should start by aligning with the PVC/Snapshot which has already solved this problem.